workflow-bot-app/src/main/java/com/symphony/bdk/workflow/configuration/WorkflowDeployer.java

```diff
@@ -80,10 +92,13 @@ public void handleFileEvent(Path changedFile, WatchEvent<Path> event) throws IOE
         this.addWorkflow(changedFile);
 
       } else if (event.kind().equals(StandardWatchEventKinds.ENTRY_DELETE)) {
-        String workflowId = deployedWorkflows.get(changedFile).getLeft();
-        this.workflowEngine.undeploy(workflowId);
-        this.deployedWorkflows.remove(changedFile);
-        this.workflowIdPathMap.remove(workflowId);
+        Optional<VersionedWorkflow> versionedWorkflow = this.versioningService.find(changedFile);
+        if (versionedWorkflow.isPresent()) {
+          String workflowId = versionedWorkflow.get().getId();
+          String workflowVersion = versionedWorkflow.get().getVersion();
+          this.workflowEngine.undeploy(workflowId);
```

Comment:
don't we check the on-going process instance before doing the undeployment?

Reply:
As discussed, we keep cascade=true in undeploy() method, we delete the pending processes without any check.